client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+                    </b-card>
+                </b-col>
+                <b-col>
+                    <b-card
+                        border-variant="primary"
+                        header="How would like to proceed?"
+                        header-bg-variant="primary"
+                        header-text-variant="white"
+                        align="center"
+                    >
+                        <b-button
+                            @click="setSharing(actions.share_with, shareWithEmail, share_option.make_public)"
+                            v-if="item.extra.can_change.length > 0"
+                            block
+                            variant="outline-primary"
+                            >Make datasets public</b-button
+                        >
+                        <b-button
+                            @click="
+                                setSharing(actions.share_with, shareWithEmail, share_option.make_accessible_to_shared)
+                            "
+                            v-if="item.extra.can_change.length > 0"
+                            block
+                            variant="outline-primary"
+                            >Share only with {{ shareWithEmail }}</b-button
+                        >
+                        <b-button
+                            @click="setSharing(actions.share_with, shareWithEmail, share_option.no_changes)"
+                            block
+                            variant="outline-primary"
+                            >Share Anyway
+                        </b-button>
+                        <b-button @click="getModel()" block variant="outline-danger">Cancel </b-button>
+                    </b-card>
+                </b-col>
+            </b-row>
+            <!--            </b-card-group>-->
+        </div>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCopy, faEdit, faUserPlus, faUserSlash } from "@fortawesome/free-solid-svg-icons";
+import { getAppRoot } from "onload/loadConfig";
+import { getGalaxyInstance } from "app";
+import SlugInput from "components/Common/SlugInput";
+import axios from "axios";
+import Multiselect from "vue-multiselect";
+import { copy } from "utils/clipboard";
+
+Vue.use(BootstrapVue);
+[faCopy, faEdit, faUserPlus, faUserSlash].forEach((icon) => library.add(icon));
+
+const defaultExtra = () => {
+    return {
+        cannot_change: [],
+        can_change: [],
+        can_share: true,
+    };
+};
+export default {
+    components: {
+        FontAwesomeIcon,
+        SlugInput,
+        Multiselect,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        plural_name: {
+            type: String,
+            required: true,
+        },
+        model_class: {
```

Comment:
are you sure? that's not really my code. Is it a rule of smth? Because we use `_` as a separator in almost every component access UI. I don't think, it's an issue, but it's ok, I will change it

Reply:
Yes, it should be camelCased.  Right now it's just a warning because there are so many of them that are broken, but this is a strongly recommended eslint rule we want to enable.

The reason they're still not all fixed is that when props are passed based on dict contents from the backend, where it can actually be snake_case that gets dictified into json, so each occasion requires examination.